utils/src/newtype.rs

```diff
@@ -15,7 +15,7 @@
 
 #[macro_export]
 macro_rules! newtype {
```

Comment:
I would call this `strongly_typed`, btw.

Reply:
This pattern is commonly known as the newtype pattern (for example the Rust Book calls it that). I prefer to keep the name `newtype`.